nemo_skills/inference/generate.py

```diff
@@ -261,10 +265,16 @@ def __init__(self, cfg: GenerateSolutionsConfig):
         self.output_lock = None
 
     def setup_llm(self):
+        self.sandbox = get_sandbox(**self.cfg.sandbox) if self.cfg.sandbox is not None else None
+
         if self.cfg.code_execution:
-            sandbox = get_sandbox(**self.cfg.sandbox) if self.cfg.sandbox is not None else None
-            llm = get_code_execution_model(**self.cfg.server, sandbox=sandbox)
-        elif self.cfg.online_genselect:
+            llm = get_code_execution_model(**self.cfg.server, sandbox=self.sandbox)
+        elif self.cfg.with_tools:
+            llm = get_tool_calling_model(**self.cfg.server, sandbox=self.sandbox)
```

Comment:
It would be better to pass the sandbox config here rather than the sandbox object, since it will need to be passed over a communication barrier

Reply:
`self.sandbox` is an a concrete instance of `class Sandbox`. But I can't pass this to a tool--it has to be something I can pass to a command line call. So it would be better if you did something like:
```
get_tool_calling_model(**self.cfg.server, sandbox_cfg=self.cfg.sandbox, tool_cfg=self.cfg.tools)
```
or
```
get_tool_calling_model(cfg=self.cfg)
```

but also, feel free to defer and I can adjust the interface once I finish the tool calling implementation (probably later today or early tomorrow).